nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -88,11 +89,24 @@ public void initialize(String name, Map<String, String> options) {
         options.get(removePrefix.apply(NessieConfigConstants.CONF_NESSIE_REF));
     String requestedHash =
         options.get(removePrefix.apply(NessieConfigConstants.CONF_NESSIE_REF_HASH));
-    NessieApiV1 api =
+
+    NessieClientBuilder<?> nessieClientBuilder =
         createNessieClientBuilder(
                 options.get(NessieConfigConstants.CONF_NESSIE_CLIENT_BUILDER_IMPL))
-            .fromConfig(x -> options.get(removePrefix.apply(x)))
-            .build(NessieApiV1.class);
+            .fromConfig(x -> options.get(removePrefix.apply(x)));
+    final String apiVersion = options.get(NessieUtil.CLIENT_API_VERSION);
+    NessieApiV1 api;
+    if (apiVersion == null || apiVersion.equalsIgnoreCase("v2")) {
+      // default version is set to v2.
+      api = nessieClientBuilder.build(NessieApiV2.class);
```

Comment:
I'd still default to v1. Otherwise, users with existing jobs will get failures because of URI / API version mismatch and will have to debug failures and adjust their configs with urgency.

Reply:
Reverted changing the default.